core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ExcludeConfigBuildItem.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.deployment.builditem.nativeimage;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * A build item that allows extension to configure the native-image compiler to effectively
+ * ignore certain configuration files in specific jars.
+ *
+ * The {@code jarFile} property specifies the name of the jar file or a regular expression that can be used to
+ * match multiple jar files.
+ *
+ * The {@code resourceName} property specifies the name of the resource file or a regular expression that can be used to
+ * match multiple resource files.
+ * For the match to work, the resources need to be part of the matched jar file(s) (see {@code jarFile})
+ *
+ * See https://github.com/oracle/graal/pull/3179 for more details.
+ */
+public final class ExcludeConfigBuildItem extends MultiBuildItem {
+
+    private final String jarFile;
+    private final String resourceName;
+
+    public ExcludeConfigBuildItem(String jarFile, String resourceName) {
```

Comment:
For these `String` version of the constructor, perhaps we should have an additional constructor which just takes the jar file name/regex and defaults the resource name to `META-INF/native-image/native-image.properties` (we need to experiment/test to see if that requires a leading `/`, but you get the idea), because most of these libraries which package these configs would be packaging it in this specific resource since that's what's used by the graalvm while looking for these configs[1]

[1] https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/metainf/NativeImageMetaInfWalker.java#L43

Reply:
Good idea, done

Reply:
Thank you. Looks good.

The only thing that I'm a bit uncertain is whether that `META-INF/...` path will need a leading `/` slash for it to work. But that part can only be confirmed by a test case. I don't think it's easy to add that test case and test the native image generation, until some processor/extension starts using this build item. So I think this is good enough for me in its current form

Reply:
Yeah, exactly.

Looking at the examples from Galder's PR, it shouldn't be needed, but like you said, we can only be sure with a proper test